datafusion/datasource/src/file_groups.rs

```diff
@@ -213,7 +213,13 @@ impl FileGroupPartitioner {
             .iter()
             .map(|f| f.object_meta.size as i64)
             .sum::<i64>();
-        if total_size < (repartition_file_min_size as i64) || total_size == 0 {
+
+        // bail if we are asked to *split* a set of files that are already too small
+        // if we are being asked to consolidate, we proceed
+        if (total_size < (repartition_file_min_size as i64)
+            && target_partitions >= file_groups.len())
+            || total_size == 0
```

Comment:
```suggestion
        if target_partitions >= file_groups.len()
            && (total_size < (repartition_file_min_size as i64) || total_size == 0)
```
I think we should proceed with consolidating empty partitions as well (i.e. total_size == 0)

Reply:
Thanks for the suggestion, I considered it - sounds tempting, but I'm afraid that 0 probably means "empty batch" or size unknown. Even if the value is actually zero, I don't think the code below makes much sense unless heavily changed.